src/Conf.ml

```diff
@@ -1009,8 +1026,11 @@ let parse_line config ~verbose ~from s =
   | _ -> Error (`Malformed s)
 
 let is_project_root dir =
-  List.exists project_root_witness ~f:(fun name ->
-      Caml.Sys.file_exists (Filename.concat dir name) )
+  match root with
+  | Some root -> Fpath.(equal (v dir |> normalize) root)
```

Comment:
It would probably be better to normalize the dir pathname when first entering collect_files (or earlier) instead of repeatedly here.